src/commands/capsule/tasks/start/system.ts

```diff
@@ -0,0 +1,82 @@
+import * as fs from 'fs';
+import { integer, string } from '../../../../lib/flags'
+import debugFn = require('debug')
+import { appendTag, createTask } from '../../../../lib/tasks'
+import { NewTask } from '../../../../lib/api'
+import { Command } from '../../../../lib/command'
+
+const debug = debugFn(`capsule:tasks:start:system`)
+
+export default class TasksStartSystemCommand extends Command {
+
+  static description = `Start a system task with the given configuration`
+  static hidden = true
+
+  static flags = { // how to change ordering of these in help command?
+    task_type_name: string({
+      char: `N`,
+      required: true,
+      multiple: false,
+      description: `Name of the task type for this task`,
+    }),
+    task_type_major: integer({
+        char: `m`,
+        required: true,
+        multiple: false,
+        description: `Major version of the task type`,
+    }),
+    task_name: string({
+        char: `n`,
+        required: true,
+        multiple: false,
+        description: `Name of the task`,
+    }),
+    assign_to: string({
+        char: `a`,
+        required: true,
+        multiple: false,
+        description: `Devices on which to start this task`,
+    }),
+    args: string({
+        char: `A`,
+        required: true,
+        multiple: false,
+        description: `Encoded JSON or @filename`,
+    }),
+    tag: string({
+        char: `t`,
+        required: false,
+        multiple: false,
+        description: `Optional tag to tie to your task`
+    })
+  }
+
+  async run(): Promise<void> {
+    const { flags } = await this.parse(TasksStartSystemCommand)
+    var encoded_string = flags.args
+    if (encoded_string.charAt(0) == '@') {
+      var encoded_string = fs.readFileSync(encoded_string.substring(1, encoded_string.length),{ encoding: 'utf8', flag: 'r' }).toString()
+    }
+    flags.args = JSON.parse(encoded_string)
+
+    if (flags.tag) {
+        flags.args = appendTag(flags.args, flags.tag)
```

Comment:
I'd like to better understand what the value of `flags.args.tags` should be. Is it a string delimited by commas or is it an array of strings?

Do we need to merge tag values from the JSON file or do we overwrite the tags value if `--tag` is provided?

This is a common merge pattern:
```
flags.args.tags = {
  ...flags.args,
  tags: [ ...flags.args.tags, tag ]
}
```

Reply:
It should be an array of strings (which I am just now realizing should by default always contain the task type name along with the specified tag). For example, when starting a delivery task in the capsule CLI:
```
$ ./capsule tasks start system delivery 1 test frog @delivery_args.json --tag myTag
```
You get (see tags in args):
```
$ ./capsule tasks list
[{"assign_to":["urn:relay-resource:name:device:frog"],"task_name":"test","subscriber_id":"1234","task_type_name":"delivery","status":"running","task_type_namespace":"system","args":{"stoppable_code":"myCode1234","tags":["myTag","delivery"],"sla":{"max_deviation":60,"target_from_start":80},"deliverers":["frog","sloth"],"task_timeout":200,"min_time_in_destination":60,"verbose":true,"destination":["tower a","1st","cafeteria"]},"task_id":"56IkWenwj9AfrXrhvnRN9A0A","workflow_id":"wf_56IkWenwj9AfrXrhvnRN9A0A_P9A25yGMmVNgPt6asjJnqoA","timestamp":"2023-09-21T19:14:11Z","workflow_instance_id":"JV2nDhAKZiQhOjqRLZsATC","task_type_major":1}]

```